### ISSUE TITLE
GenomicSequences.java is not Serializable

### ISSUE THREAD

**Issue by emreatbina** [pcingola/SnpEff#85]
Even though ca.mcgill.mcb.pcingola.snpEffect.SnpEffectPredictor class is Serializable, it is not possible to serialize it because, it has a reference to ca.mcgill.mcb.pcingola.interval.Genome, and that has a reference to ca.mcgill.mcb.pcingola.binseq.GenomicSequences which is not Serializable. Making GenomicSequences Serializable would allow the entire SnpEffectPredictor object to be serializable, which can then be used for running SnpEff with tools such as Apache Spark where the expensive SnpEffectPredictor object can be created once and shared in memory, instead of being created inside every worker.

**Comment by pcingola** [comment]
Great. I'll try to get it done when I get back home (I'm out of town now).

**Comment by emreatbina** [comment]
@pcingola Thank you very much.

**Comment by pcingola** [comment]
PR merged. Thank you!

**Comment by heuermh** [comment]
@emreatbina have you succeeded in running SnpEff on Apache Spark?  I was about to pursue the same in the context of the https://github.com/bigdatagenomics/adam project.

**Comment by pcingola** [comment]
I think they did.
I fixed this some time ago, but let me know if you need to Serialize another class.

**Comment by emreatbina** [comment]
Hi @heuermh, we have succeeded to run SnpEff on Apache Spark after this PR was merged, but we ran into other performance problems mostly due to memory usage. More specifically, the SnpEffectPredictor object can get quite big and sharing or broadcasting it across all Spark executors can cause some memory issues. The other option is to create this object inside every executor, but that means that every executor has to take the time (order of minutes) to create this object first before moving on to effect prediction. Executors must also have enough memory to keep this object in memory.
We haven't really got around to it yet, but it would be nice to add a layer of abstraction to SnpEff so that it can also persist SnpEffectPredictor to somewhere like HDFS or HBase when necessary, and use that during effect prediction. 
I would be happy to talk more about it if you are interested.
